### PR TITLE
feat: DCMAW-7860 Automate Versioning

### PR DESCRIPTION
**Pull request by MarieseMikely**
# feat: DCMAW-7860 Automate Versioning

Update pull request.yml file to include validate PR name job and increment version job to the quality report.yml file 

# Checklist

## Before raising your pull request:
- [x] Commit messages that conform to conventional commit messages
- [ ] ~Ran the app locally ensuring it builds~ 
- [ ] ~Ran the tests locally ensuring they pass on Build~
- [x] Pull request has a clear title with a short description about the feature or update
- [ ] ~Created a `draft` pull request if it is not yet ready for review~

## Before your pull request can be reviewed:
- [ ] ~Met all of the acceptance criteria specified in the user story on Jira~
- [ ] ~Reviewed your own code to ensure you are following the style guidelines~
- [ ] ~Ran the app and tested the feature on a range of device sizes
      Please include iPod Touch, iPhone SE and iPhone 11 as a minimum.~
- [ ] ~Written Unit and Integration tests if needed~

- [ ] Met all accessibility requirements?
    - [ ] ~Checked dynamic type sizes are applied~
    - [ ] ~Checked VoiceOver can navigate your new code~
    - [ ] ~Checked a user can navigate only using a keyboard around your new code~ 

## Before merging your pull request:
- [ ] ~Ensure that the code coverage and SonarCloud checks have passed~
- [ ] ~Actioned and resolved all comments, reaching out to reviewers for clarifications if necessary.~
- [ ] ~Ran the app to ensure that no regressions have been caused by changes during code review.~

### CODE REVIEW

File: .github/workflows/pull-request.yml [hunk 1]
```diff
@@ -28,6 +28,12 @@ jobs:
       - name: Run Linter
         run: |
           swiftlint --strict
+          
+      - name: Validate Pull Request Name
+        id: versioning
+        uses: Oliver-Binns/Versioning@main
```
Comment (sandy-may) [inline .github/workflows/pull-request.yml:34]: @Oliver-Binns should we be pinning to main or a specific commit SHA? I _think_ this goes against the GDS way but I can't find the right docs. 

There's a chance the maintainer of this package deploys a breaking change/defect

File: .github/workflows/quality-report.yml [hunk 2]
```diff
@@ -36,3 +38,10 @@ jobs:
         env:
           SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          
+      - name: Increment Version
+        id: versioning
+        uses: Oliver-Binns/Versioning@v0.2.1
```
Comment (sandy-may) [inline .github/workflows/quality-report.yml:44]:
```suggestion
        uses: Oliver-Binns/Versioning@0.2.1
```

Comment (MarieseMikely) [reply .github/workflows/quality-report.yml]: using `@1.0.0` for the Versioning repo

File: .github/workflows/pull-request.yml [hunk 3]
```diff
@@ -28,6 +28,12 @@ jobs:
       - name: Run Linter
         run: |
           swiftlint --strict
+          
+      - name: Validate Pull Request Name
+        id: versioning
+        uses: Oliver-Binns/Versioning@1.0.0
```
Comment (amopatel1234) [inline .github/workflows/pull-request.yml:34]: should this not be `Oliver-Binns/Versioning@v1`

File: .github/workflows/quality-report.yml [hunk 4]
```diff
@@ -36,3 +38,10 @@ jobs:
         env:
           SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          
+      - name: Increment Version
+        id: versioning
+        uses: Oliver-Binns/Versioning@1.0.0
```
Comment (amopatel1234) [inline .github/workflows/quality-report.yml:44]: same as the above, should this not be `Oliver-Binns/Versioning@v1`

Comment (MarieseMikely) [reply .github/workflows/quality-report.yml:44]: @sandy-may left a comment stating the `v` in the annotation should be removed to reference the release tag correctly.

**Review by amopatel1234 — APPROVED** [review]

**Pull request merged**